code/game/objects/effects/weeds.dm

```diff
@@ -145,18 +143,79 @@
 
 /obj/effect/alien/weeds/node/update_icon()
 	overlays.Cut()
-	overlays += "weednode"
+	overlays += node_icon
 
-/obj/effect/alien/weeds/node/Initialize(mapload, obj/effect/alien/weeds/node/node, mob/living/carbon/xenomorph/X)
+/obj/effect/alien/weeds/node/Initialize(mapload, obj/effect/alien/weeds/node/node)
 	for(var/obj/effect/alien/weeds/W in loc)
 		if(W != src)
 			qdel(W) //replaces the previous weed
 			break
+	. = ..()
 
-	overlays += "weednode"
-	. = ..(mapload, src)
+	update_icon()
 
 	// Generate our full graph before adding to SSweeds
 	node_turfs = filled_turfs(src, node_range, "square")
 	SSweeds.add_node(src)
+
+
+// =================
+// stronger weed node
+/obj/effect/alien/weeds/node/strong
+	name = "strong purple sac"
+	desc = "A weird, pulsating node. This looks pretty tough."
+	node_range = NODERANGE*2
+	max_integrity = 120
+
+
+// =================
+// hivemind core
+/obj/effect/alien/weeds/node/hivemindcore
+	name = "hivemind core"
+	desc = "A very weird, pulsating node. This looks almost alive."
+	max_integrity = 600
+	node_icon = "weed_hivemind4"
+	var/mob/living/carbon/xenomorph/hivemind/parent
+
+/obj/effect/alien/weeds/node/hivemindcore/Initialize(mapload, obj/effect/alien/weeds/node/node)
+	. = ..()
+	parent = new(loc)
+	set_light(7, 5, LIGHT_COLOR_PURPLE)
+
+/obj/effect/alien/weeds/node/hivemindcore/Destroy()
+	set_light(0, 0, null)
```

Comment:
Aren't lights properly removed on Destroy by default?

Reply:
Not that i could see

Reply:
It's pretty obscure but we have this:
```
/atom/Destroy()
	QDEL_NULL(light)
```
which leads to this:
```
/datum/light_source/Destroy(force)
```
So lights already get removed properly on Destroy.

Reply:
Alright will remove this.